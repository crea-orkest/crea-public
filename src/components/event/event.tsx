import { EventListItem } from 'components/eventListItem'
import React from 'react'
import { getEvent } from '../../graphql/getters/getEvent'

export interface Props {
  className?: string
  id: string
  size?: 'small' | 'large'
  isLast?: boolean
}

export const Event: React.FC<Props> = async ({
  className,
  id,
  size = 'small',
  isLast,
}: Props) => {
  const { data } = await getEvent({ id })
  if (!data) return null

  return (
    <EventListItem
      className={className}
      data={data}
      size={size}
      isLast={isLast}
    />
  )
}
