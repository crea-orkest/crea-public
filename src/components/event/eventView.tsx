import type { Event } from 'graphql/types/event'
import Image from 'next/image'
import React from 'react'
import classNames from 'classnames'
import styles from './styles.module.scss'

export interface Props {
  data: Event
}

export const EventView: React.FC<Props> = ({ data }: Props) => {
  return (
    <div className={classNames(styles.card)}>
      {data.image?.url && (
        <Image
          priority={true} // TODO: only for the first item
          className={classNames(styles.card__image)}
          src={data.image.url}
          alt={data.image.description}
          width={data.image.width ?? 100}
          height={data.image.height ?? 100}
        />
      )}
      <h2>{data.title}</h2>
      {data.persons && data.persons.length > 0 && (
        <ul>
          {data.persons.map((person) => {
            return (
              <li key={person.id}>
                {person.name} | {person.description}
              </li>
            )
          })}
        </ul>
      )}
      {data?.locations?.map((item) => {
        if (!item.id) return
        return (
          <div key={item.id} className={classNames(styles.card__locations)}>
            <p>{item.id}</p>
            {item.startTime && <p>{new Date(item.startTime).toDateString()}</p>}
          </div>
        )
      })}
    </div>
  )
}
