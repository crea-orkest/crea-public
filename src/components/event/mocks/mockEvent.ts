import type { Event } from 'graphql/types/event'

export const mockEvent: Event = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [{ startTime: 'DATETIME', id: 'test id' }],
  persons: [],
}
