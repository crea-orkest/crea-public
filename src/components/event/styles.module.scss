.card {
  margin: 3rem 0 0 0;
  background-color: antiquewhite;

  &__image {
    width: 100%;
    height: 20rem;
    object-fit: cover;
  }

  &__locations {
    display: grid;
    gap: 1rem;
    grid-template-columns: 1fr;

    // TODO: use variables or tailwind?
    @media screen and (min-width: 48rem) {
      grid-template-columns: 1fr 1fr;
    }
  }
}
