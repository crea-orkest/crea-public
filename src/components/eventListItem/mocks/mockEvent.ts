import type { Event } from 'graphql/types/event'

export const mockEvent: Event = {
  __typename: 'ConcertRecord',
  _createdAt: '2023-09-01T13:32:36+01:00',
  _firstPublishedAt: undefined,
  _publishedAt: undefined,
  _updatedAt: '2023-08-14T09:37:24+01:00',
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [{ startTime: '2023-09-01T13:32:36+01:00', id: 'test id' }],
  persons: [],
  program: [],
  url: '/some-slug',
  slug: 'some-slug',
  content: [],
}
