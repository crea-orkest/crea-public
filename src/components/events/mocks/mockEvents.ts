import type { Event } from 'graphql/types/event'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  url: '/concerten/concert-1',
  image: undefined,
  locations: [],
  persons: [],
}
