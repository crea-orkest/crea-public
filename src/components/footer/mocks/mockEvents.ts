import { type Event } from 'graphql/types/event'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  image: undefined,
  locations: [],
  persons: [],
  url: '/some-slug',
  content: [],
}
