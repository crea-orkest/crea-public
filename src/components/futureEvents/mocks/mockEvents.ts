import { type Event } from 'graphql/types/event'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  image: undefined,
  locations: [],
  persons: [],
  program: [],
  url: '/some-slug',
  content: [],
}
