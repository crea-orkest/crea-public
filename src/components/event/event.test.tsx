import { Event } from './event'
import type { Props } from './event'
import React from 'react'
import { getEvent } from 'graphql/getters/getEvent'
import { mockEvent } from './mocks/mockEvent'
import { resolvedComponent } from '../../utils/testHelpers/resolvedComponent'
import { render, screen } from '@testing-library/react'

jest.mock('../location/location', () => {
  const originalModule = jest.requireActual('../location/location')
  return {
    __esModule: true,
    ...originalModule,
    Location: jest.fn(() => <p>Location Component</p>),
  }
})

jest.mock('../../graphql/getters/getEvent', () => {
  const originalModule = jest.requireActual('../../graphql/getters/getEvent')
  return {
    __esModule: true,
    ...originalModule,
    getEvent: jest.fn(),
  }
})

const getEventMock = jest.mocked(getEvent)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getEventMock.mockResolvedValueOnce({
      data: mockEvent,
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(Event, {
      id: 'some id',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()

    expect(screen.getByText('name')).toBeTruthy()
    expect(screen.getByText('Location Component')).toBeTruthy()
  })
})
