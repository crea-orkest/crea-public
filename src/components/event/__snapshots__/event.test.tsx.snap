// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concert component shows all the data 1`] = `
<div>
  <div
    class="card"
  >
    <h2>
      name
    </h2>
    <div
      class="card__locations"
    >
      <p>
        Location Component
      </p>
    </div>
  </div>
</div>
`;
